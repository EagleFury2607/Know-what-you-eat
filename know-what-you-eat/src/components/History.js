import React, { Component } from "react";
import Axios from "axios";
import NavbarComponent from "./NavbarComponent";
import "./css/history.css";
class History extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      data: [],
    };
    this.fetchHistory = this.fetchHistory.bind(this);
  }
  componentDidMount() {
    // console.log("Component is mounted !");
    this.fetchHistory();
  }
  fetchHistory = () => {
    const jwt = localStorage.getItem("jwt");
    if (!jwt) {
      this.props.history.push("/login");
    }
    Axios.get("/getHistory", {
      headers: {
        Authorization: `Bearer ${jwt}`,
      },
      option: {
        "Access-Control-Allow-Origin": "*",
      },
    })
      .then((result) => {
        this.setState({
          data: result.data[0][0].history,
        });
        // console.log(this.state.data);
      })
      .catch((err) => {
        console.log("ERR", err);
      });
  };
  render() {
    var printOut = "";
    if (this.state.data !== []) {
      console.log(this.state.data);
      printOut = this.state.data.map(function (data, id) {
        return (
          <div>
            <li key={id}>
              Date: {data.date}
              <br></br>
              Front Fruit: {data.frontFruitName}
              <br></br>
              Side Fruit: {data.sideFruitName}
              <br></br>
              Time Taken: {data.time}
              <br></br>
              Calorie: {data.calorie}
            </li>
          </div>
        );
      });
    }
    return (
      <div>
        <NavbarComponent userLogginStatus={"Logout"} />
        <h1>HISTORY</h1>
        <div
          id="showData"
          style={{
            display: "flex",
            alignItems: "center",
            flexDirection: "column",
          }}
        >
          <div className="list-type1">
            <ol>{printOut}</ol>
          </div>
        </div>
      </div>
    );
  }
}

export default History;
