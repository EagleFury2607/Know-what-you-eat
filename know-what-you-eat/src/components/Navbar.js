import React,{Component} from 'react';
import {FaAmazonPay} from 'react-icons/fa'
import './css/navbar.css';

export default class App extends Component {
    state = {
        toggle:false
    }
    Toggle = () => {
        this.setState({toggle:!this.state.toggle})
    }
    render() {
        return (
          <div>
              <div className="navBar">
                    <button onClick={this.Toggle}>
                        {/* <FaAlignRight /> */}
                        <FaAmazonPay />
                    </button>
                    <ul className={this.state.toggle ? "nav-links show-nav" : "nav-links"}>
                        <li href="#">Home</li>
                        <li href="#">About us</li>
                        <li href="#">Contact</li>
                    </ul>
              </div>
          </div>
        );
    }
}