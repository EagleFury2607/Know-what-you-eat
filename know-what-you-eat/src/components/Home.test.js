import React from "react";
import History from "./History";
import Navbar from "react-bootstrap/Navbar";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const component = renderer.create(<History />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
