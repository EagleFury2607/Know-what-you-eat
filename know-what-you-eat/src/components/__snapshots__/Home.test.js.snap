exports[`test renders without crashing 1`] = `
<div>
  Hello world !
</div>
`;
